google/cloud/internal/async_read_write_stream_tracing_test.cc

```diff
@@ -321,6 +347,41 @@ TEST(AsyncStreamingReadWriteRpcTracing, SpanEndsOnDestruction) {
   EXPECT_THAT(spans, ElementsAre(SpanNamed("span")));
 }
 
+TEST(AsyncStreamingReadWriteRpcTracing,
+     UnstartedStreamShouldNotExtractMetadata) {
+  auto span_catcher = testing_util::InstallSpanCatcher();
+
+  {
+    auto mock = std::make_unique<MockStream>();
+    auto span = MakeSpan("span");
+    auto context = std::make_shared<grpc::ClientContext>();
+    TestedStream stream(context, std::move(mock), span);
+  }
+
+  auto spans = span_catcher->GetSpans();
+  EXPECT_THAT(spans, ElementsAre(SpanNamed("span")));
+}
+
+TEST(AsyncStreamingReadWriteRpcTracing, StartedStreamShouldExtractMetadata) {
+  auto span_catcher = testing_util::InstallSpanCatcher();
+  {
+    auto span = MakeSpan("span");
+    auto mock = std::make_unique<MockStream>();
+    auto context = std::make_shared<grpc::ClientContext>();
+    EXPECT_CALL(*mock, Start).WillOnce([context] {
+      SetServerMetadata(*context, {});
```

Comment:
I think I would say:
```cc
SetServerMetadata(*context, {{"key", "value"}})
```

And then later check for:
```cc
SpanHasAttributes(OTelAttribute<std::string>("rpc.grpc.response.metadata.key", "value")
```

Reply:
Done.